### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by Swaagie**
## Summary

Update documentation

## Changelog

Done

## Test Plan

NA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,45 @@
...
-- Automatically [updating built assets](#auto-update-of-builds) when new versions of dependent modules are published.
+- Serving [fully built assets](#builds) (e.g. JavaScript, CSS, etc) for
+  published modules through [a CDN target](#cdn-support).
+- Decoupling [upgrade and rollback](#upgrade-and-rollback) of individual
+  dependencies across multiple environments through use of `npm dist-tag`.
+- Automatically [updating built assets](#auto-update-of-builds) when new
+  versions of dependent modules are published.
 
-In other words the Warehouse is designed to give as many programmatic guarantees that it is safe to "always be on latest" and make rolling back as painless as possible when issues arise.
+In other words the Warehouse is designed to give as many programmatic
```
Comment (msluther) [inline README.md]: `the Warehouse` -> `Warehouse.ai`

File: README.md [hunk 2]
```diff
@@ -60,14 +90,15 @@ npm dist-tag add module@version prod
 ```
 ![Diagram of npm dist-tag promotion of modules](assets/promote.png)
 
-Warehouse.ai builds are an interaction between multiple smaller microservices to garantuee high
-concurrency and stability.
+Warehouse.ai builds are an interaction between multiple smaller microservices
+to garantuee high concurrency and stability.
```
Comment (msluther) [inline README.md]: `garantuee` -> `guarantee`

File: README.md [hunk 3]
```diff
@@ -120,15 +162,21 @@ And we wish to rollback production to `1.0.4` then:
 npm dist-tag add my-module@1.0.4 prod
 ```
 
-This will trigger a build of `my-module@1.0.4` (since it's dependencies may have changed since 1.0.4 was latest in production) and a build of all modules that depend on `my-module`.
+This will trigger a build of `my-module@1.0.4` (since it's dependencies may
+have changed since 1.0.4 was latest in production) and a build of all modules
+that depend on `my-module`.
```
Comment (msluther) [inline README.md]: This paragraph isn't really accurate (can be updated later since you're just touching formatting). but if 1.0.4 was in prod before, it'll just reuse the same build as last time.

File: README.md [hunk 4]
```diff
@@ -120,15 +162,21 @@ And we wish to rollback production to `1.0.4` then:
...
 
 ### Auto-update of builds
 
 > Stability: 1 – **Unstable**
 
-The first (and most important) point developers need to be aware of is that builds from the Warehouse are always on latest version of private dependencies tagged with that particular environment.
+The first (and most important) point developers need to be aware of is that
+builds from the Warehouse are always on latest version of private dependencies
```
Comment (msluther) [inline README.md]: `always` is a strong word. it doesn't constantly update existing builds, you only get latest semver matching versions on new builds, not for existing builds.

Comment (msluther) [reply README.md]: again, can be addressed later since you're just updating formatting ;)

File: README.md [hunk 5]
```diff
@@ -194,28 +255,41 @@ POST /builds/:pkg                     # Ad-hoc build
...
+```scala
+PATCH /promote/:pkg/:env/:version     # Promote a package@version to an environment
+```
+
 ### Environment-specific installation
 
-Warehouse allows for installation against a specific `dist-tag` via the `REGISTRY-ENVIRONMENT` header. Although `npm` does not allow for headers to be set directly, [carpenter] sets these headers internally during it's install process.
+Warehouse allows for installation against a specific `dist-tag` via the
+`REGISTRY-ENVIRONMENT` header. Although `npm` does not allow for headers to be
+set directly, [carpenter] sets these headers internally during it's install
```
Comment (fritzmonkey) [inline README.md]: This link is broken. Should be `[carpenterd]`
